src/map.cpp

```diff
@@ -37,7 +37,12 @@ Map::Map(const std::string &name, const SizedType &type, const MapKey &key, int
   if (key_size == 0)
     key_size = 8;
 
-  if (type.IsCountTy() && !key.args_.size())
+  if (type.IsBufferTy())
+  {
+    map_type_ = BPF_MAP_TYPE_ARRAY;
```

Comment:
doesn't this break cases like `@thing[pid] = buf(1,1)`?

Reply:
it does :-\ I'll add separate type for that

Reply:
Can you add a test case for the buf thing too?